File: templates/template.py
import argparse


def part1(input):
    return -1


def part2(input):
    return -1


if __name__ == "__main__":
    arg_parser = argparse.ArgumentParser()
    arg_parser.add_argument(
        "--sample",
        "-s",
        action="store_true",
        help="Use sample.txt (manually filled in)",
    )
    args = arg_parser.parse_args()
    input_file = "input.txt"
    if args.sample:
        print("USING SAMPLE INPUT FILE")
        input_file = "sample.txt"

    parsed_input = None  # Use some parsing method here

    print("Part 1:", part1(parsed_input))
    print("Part 2:", part2(parsed_input))

File: templates/template.go
package main

import (
	"bufio"
	"flag"
	"fmt"
	"os"
)

func part1(input any) int {
	return 0
}

func part2(input any) int {
	return 0
}

func parseInput(filename string) (any, error) {
	file, err := os.Open(filename)
	if err != nil {
		return nil, err
	}
	defer file.Close()

	var parsed []string
	scanner := bufio.NewScanner(file)

	// Use any parsing method here
	for scanner.Scan() {
		parsed = append(parsed, scanner.Text())
	}

	if err := scanner.Err(); err != nil {
		return nil, err
	}

	return parsed, nil
}

func main() {
	useSample := flag.Bool("sample", false, "Use sample.txt (manually filled in)")
	flag.Parse()
	fileName := "input.txt"
	if *useSample {
		fmt.Println("USING SAMPLE INPUT FILE")
		fileName = "sample.txt"
	}
	input, err := parseInput(fileName)
	if err != nil {
		fmt.Println("Error reading input:", err)
		return
	}
	fmt.Println("Part 1:", part1(input))
	fmt.Println("Part 2:", part2(input))
}
